fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -391,7 +391,35 @@ public PhysicalPlanTranslator(ColumnRefFactory columnRefFactory) {
         }
 
         public PlanFragment translate(OptExpression optExpression, ExecPlan context) {
-            return visit(optExpression, context);
+            PlanFragment fragment = visit(optExpression, context);
+            computeFragmentCost(context, fragment);
+            return fragment;
+        }
+
+        private void computeFragmentCost(ExecPlan context, PlanFragment fragment) {
+            for (PlanFragment child : fragment.getChildren()) {
+                computeFragmentCost(context, child);
+            }
+            OptExpression output = getOptExpression(context, fragment.getPlanRoot());
+
+            // scan fragment
+            if (fragment.getLeftMostNode() instanceof ScanNode) {
+                fragment.setFragmentCost(output.getCost());
+                return;
+            }
+
+            List<OptExpression> inputs = fragment.getChildren().stream().map(PlanFragment::getPlanRoot)
+                    .map(p -> getOptExpression(context, p)).collect(Collectors.toList());
+
+            double childCost = inputs.stream().map(OptExpression::getCost).reduce(Double::sum).orElse(0D);
+            fragment.setFragmentCost(output.getCost() - childCost);
```

Comment:
Fragment cost is means the operator cost in this fragment. If its a colocate join fragment, we just need take the top planNode cost. If it's a remote fragment, we cannot use the top planNode cost to describe this fragment cost because this cost is a sum of all optExpression tree. So we need to use this sum cost minus it's children fragment cost to obtain this fragment cost.